src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,135 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties            as ModuleProperties
+import Algebra.Module.Morphism.Structures   as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.Product
+open import Relation.Binary.Reasoning.MultiSetoid
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module PA = ModuleProperties modA
+module PB = ModuleProperties modB
+open CommutativeRing ring renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f0≈0 : f A.0ᴹ B.≈ᴹ B.0ᴹ
+  f0≈0 = begin⟨ B.≈ᴹ-setoid ⟩
+    f A.0ᴹ           ≈⟨ ⟦⟧-cong (A.≈ᴹ-sym (A.*ₗ-zeroˡ A.0ᴹ)) ⟩
+    f (0# A.*ₗ A.0ᴹ) ≈⟨ *ₗ-homo 0# A.0ᴹ ⟩
+    0# B.*ₗ f A.0ᴹ   ≈⟨ B.*ₗ-zeroˡ (f A.0ᴹ) ⟩
+    B.0ᴹ             ∎
+
+  x≈0⇒fx≈0 : ∀ {x} → x A.≈ᴹ A.0ᴹ → f x B.≈ᴹ B.0ᴹ
+  x≈0⇒fx≈0 {x} x≈0 = begin⟨ B.≈ᴹ-setoid ⟩
+    f x    ≈⟨ ⟦⟧-cong x≈0 ⟩
+    f A.0ᴹ ≈⟨ f0≈0 ⟩
+    B.0ᴹ   ∎
+
+  fx≉0⇒x≉0 : ∀ {x} → f x B.≉ᴹ B.0ᴹ → x A.≉ᴹ A.0ᴹ
+  fx≉0⇒x≉0 = contraposition x≈0⇒fx≈0
+
+  -- Zero is a unique output of linear map ≉ `const 0`.
+  zero-unique : ∀ {x} →
```

Comment:
This is an awkward way to phrase uniqueness (constructively). Normally what you do is assume you have something else (call it `z`) that behaves like a zero, and then derive that it must be equal to 0.

Reply:
Hi @JacquesCarette ,

Are you proposing a change to:

1. the type signature,
2. the definition, or
3. both?

Reply:
I think both. I'd expect something more positive, in the sense of "assume this thing z behaves just like 0, then it must in fact be equal to 0".  That's what 'uniqueness' brings to mind for me. You've got some kind of double-negative version of that.

Reply:
Oh, except that I'm going the other way in this lemma.
I'm trying to prove that if the argument to `f` is _not_ zero then the result of `f` cannot be zero:

```agda
x A.≉ᴹ A.0ᴹ → f x B.≉ᴹ B.0ᴹ
```

Perhaps, my function name is misleading?